src/iterator.h

```diff
@@ -8,18 +8,29 @@
 
 typedef struct iterator iterator;
 
-typedef void (*iterator_get_curr_fn)(iterator *itor,
-                                     key      *curr_key,
-                                     message  *msg);
-typedef platform_status (*iterator_at_end_fn)(iterator *itor, bool *at_end);
-typedef platform_status (*iterator_advance_fn)(iterator *itor);
+// for seek
+typedef enum comparison {
+   less_than,
+   less_than_or_equal,
+   greater_than,
+   greater_than_or_equal,
+} comparison;
+
+typedef void (*iterator_curr_fn)(iterator *itor, key *curr_key, message *msg);
+typedef bool (*iterator_in_range_fn)(iterator *itor);
+typedef platform_status (*iterator_step_fn)(iterator *itor);
+typedef platform_status (*iterator_seek_fn)(iterator  *itor,
+                                            key        seek_key,
+                                            comparison from_above);
 typedef void (*iterator_print_fn)(iterator *itor);
 
 typedef struct iterator_ops {
    /* Callers should not modify data pointed to by *key or *data */
-   iterator_get_curr_fn get_curr;
-   iterator_at_end_fn   at_end;
-   iterator_advance_fn  advance;
+   iterator_curr_fn     curr;
+   iterator_in_range_fn in_range;
```

Comment:
Break into `can_next`, `can_prev`, and implement an `iterator_can_curr` method in this file.